tests/controller_test/collect-data_test.go

```diff
@@ -0,0 +1,172 @@
+package controller_test
+
+import (
+	"context"
+	"github.com/litmuschaos/chaos-exporter/controller"
+	"github.com/litmuschaos/chaos-exporter/pkg/clients"
+	"github.com/litmuschaos/chaos-operator/api/litmuschaos/v1alpha1"
+	litmusFakeClientSet "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"testing"
+)
+
+func TestGetResultList(t *testing.T) {
+	FakeChaosNameSpace := "Fake Namespace"
+	FakeEngineName := "Fake Engine"
+
+	tests := map[string]struct {
+		chaosresult *v1alpha1.ChaosResult
+		monitoring  *controller.MonitoringEnabled
+		isErr       bool
+	}{
+		"Test Positive-1": {
+			chaosresult: &v1alpha1.ChaosResult{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      FakeEngineName,
+					Namespace: FakeChaosNameSpace,
+				},
+				Spec: v1alpha1.ChaosResultSpec{
+					ExperimentName: "exp-1",
+					EngineName:     FakeEngineName,
+				},
+			},
+			isErr: false,
+			monitoring: &controller.MonitoringEnabled{
+				IsChaosResultsAvailable: true,
+			},
+		},
+		"Test Negative-1": {
+			chaosresult: &v1alpha1.ChaosResult{},
+			isErr:       true,
+			monitoring: &controller.MonitoringEnabled{
+				IsChaosResultsAvailable: true,
+			},
+		},
+		"Test Negative-2": {
+			isErr:      true,
+			monitoring: &controller.MonitoringEnabled{},
+		},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			client := CreateFakeClient(t)
+			if !mock.isErr {
+				_, err := client.LitmusClient.LitmuschaosV1alpha1().ChaosResults(mock.chaosresult.Namespace).Create(context.Background(), mock.chaosresult, metav1.CreateOptions{})
```

Comment:
Can we use execFunc for this purpose as it provides more flexibility? We should avoid adding per testcase conditions in common test code/loop